main.tf

```
@@ -321,5 +321,5 @@ locals {
 
   # user_attribute_update_settings
   # As default, all auto_verified_attributes will become attributes_require_verification_before_update
-  user_attribute_update_settings = var.user_attribute_update_settings == null ? (length(var.auto_verified_attributes) > 0 ? { attributes_require_verification_before_update = var.auto_verified_attributes } : {}) : var.user_attribute_update_settings
+  user_attribute_update_settings = length(var.attributes_require_verification_before_update) > 0 ? { attributes_require_verification_before_update = var.attributes_require_verification_before_update } : {}
```

This is not the same logic as it was before. This new version does not care about the `auto_verified_attributes` setting at all. 

I understand you would like to simplify the settings, so taking that, and the default into account, it should be like (the code is not tested)
```suggestion
  user_attribute_update_settings = var.attributes_require_verification_before_update == null ?
    (length(var.auto_verified_attributes) > 0 ? { attributes_require_verification_before_update = var.auto_verified_attributes } : {}) :
    ( length(var.attributes_require_verification_before_update) > 0 ? { attributes_require_verification_before_update = var.attributes_require_verification_before_update } : {} )
```

Given the security implications, I would rather require extra verification, and explicit opt-out here, so we err on the safe side.